proposals/lip-0052.md

```diff
@@ -1619,27 +1699,30 @@ def emitFailedDestroyEvent(
 
 #### lock
 
-This function locks an NFT to a given module.
+This function locks an NFT to a given module. Note that an NFT can not be locked to the NFT module.
 
 ```python
 def lock(
     module: Module,
     nftID: NFTID
 ) -> None:
 
-    if NFTStore[nftID] does not exist:
-        emitFailedLockEvent(module, nftID, RESULT_NFT_DOES_NOT_EXIST)
-        raise Exception("NFT substore entry does not exist")
+    nft = getNFT(nftID)
```

Comment:
I think we should handle the possible exception from this function by emitting an event with `RESULT_NFT_DOES_NOT_EXIST` if the call to `getNFT` fails.

Reply:
Done in [this commit](https://github.com/LiskHQ/lips/pull/477/commits/876ae2c110612053e85a0c54fb65574a7e1b4bc1).